src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -113,22 +113,29 @@ private static SearchParameterInfo GetOrCreateSearchParameterInfo(SearchParamete
             IModelInfoProvider modelInfoProvider)
         {
             var issues = new List<OperationOutcomeIssue>();
-            var searchParameters = searchParamCollection.ToList();
+            var searchParameters = searchParamCollection.Select((x, entryIndex) =>
+            {
+                try
+                {
+                    return new SearchParameterWrapper(x);
+                }
+                catch (ArgumentException)
+                {
+                    AddIssue(Core.Resources.SearchParameterDefinitionInvalidResource, entryIndex);
+                    return null;
+                }
+            }).ToList();
```

Comment:
>ToList [](start = 15, length = 6)

I would filter .Where(x=>x!=null) to simplify code below.

Reply:
I think I also initially tried this, the parsing code below respecting the index of the entry in the Bundle. Null is only used when there is an entry in the Bundle that isn't a SearchParameter resource. I made this change to to avoid parsing into SearchParameterWrapper multiple times